tool/tsh/db.go

```diff
@@ -398,6 +410,30 @@ func needRelogin(cf *CLIConf, tc *client.TeleportClient, database *tlsca.RouteTo
 	return mfaRequired, nil
 }
 
+// dbUserHasChanged  checks if cf.DatabaseUser username matches the username in the database certificate.
+func dbUserHasChanged(cf *CLIConf, database *tlsca.RouteToDatabase, certPath string) (bool, error) {
+	if database.Protocol != defaults.ProtocolMongoDB {
```

Comment:
Why is this MongoDB specific? We allow specifying `--db-user` for other protocols too. In fact, I'm thinking making it mandatory for new protocols (like, SQL Server), makes things easier.

Also, what about `--db-name`? It can change also, right?

Reply:
Right now only mongo user name is used from TLS cert. For other protocols user name and database name are passed in DSN string. But yes, I agree -  this current code won't scale and the logic will need to be verified and aligned for each protocol thus I have reworked it to this logic for each case.